pkg/cri/server/image_pull.go

```diff
@@ -506,6 +508,22 @@ func (c *criService) encryptedImagesPullOpts() []containerd.RemoteOpt {
 	return nil
 }
 
+func (c *criService) authConfigOpts(auth *runtime.AuthConfig) containerd.RemoteOpt {
+	snapshotOpt := snapshots.WithAuthConfig(auth.Username,
+		auth.Password,
+		auth.Auth,
+		auth.ServerAddress,
+		auth.IdentityToken,
+		auth.RegistryToken)
+
+	unpackOpt := func(_ context.Context, uc *containerd.UnpackConfig) error {
+		uc.SnapshotOpts = append(uc.SnapshotOpts, snapshotOpt)
```

Comment:
Is this really passed to the snapshotter? IIUC `containerd.UnpackConfig.SnapshotOpts` isn't used by `*containerd.unpacker`.